securesignals-ima-devapp/src/main/java/com/uid2/dev/MainActivity.java

```diff
@@ -0,0 +1,283 @@
+                                        // AdEventType.CONTENT_PAUSE_REQUESTED is fired when you
+                                        // should pause your content and start playing an ad.
+                                        pauseContentForAds();
+                                        break;
+                                    case CONTENT_RESUME_REQUESTED:
+                                        // AdEventType.CONTENT_RESUME_REQUESTED is fired when the ad
+                                        // you should play your content.
+                                        resumeContent();
+                                        break;
+                                    case ALL_ADS_COMPLETED:
+                                        // Calling adsManager.destroy() triggers the function
+                                        // VideoAdPlayer.release().
+                                        adsManager.destroy();
+                                        adsManager = null;
+                                        break;
+                                    case CLICKED:
+                                        // When the user clicks on the Learn More button, the IMA SDK fires
+                                        // this event, pauses the ad, and opens the ad's click-through URL.
+                                        // When the user returns to the app, the IMA SDK calls the
+                                        // VideoAdPlayer.playAd() function automatically.
+                                        break;
+                                    default:
+                                        break;
+                                }
+                            }
+                        });
+                    AdsRenderingSettings adsRenderingSettings =
+                        ImaSdkFactory.getInstance().createAdsRenderingSettings();
+                    adsManager.init(adsRenderingSettings);
+                }
+            });
+
+        // When the play button is clicked, request ads and hide the button.
+        playButton = findViewById(R.id.playButton);
+        playButton.setOnClickListener(
+            view -> {
+                videoPlayer.setVideoPath(SAMPLE_VIDEO_URL);
+                requestAds(SAMPLE_VAST_TAG_URL);
+                view.setVisibility(View.GONE);
+            });
+    }
+
+    private void loadUID2Identity() {
+
+        InputStream is = getResources().openRawResource(R.raw.uid2identity);
+        StringBuilder text = new StringBuilder();
+
+        try (BufferedReader br = new BufferedReader(new InputStreamReader(is))) {
+            String line;
+
+            while ((line = br.readLine()) != null) {
+                text.append(line);
+                text.append('\n');
+            }
+
+            String jsonString = text.toString();
+
+            JSONObject jsonObject = (JSONObject) new JSONTokener(jsonString).nextValue();
+
+            // Emulate A UID2Identity With Valid Times
```

Comment:
Nit: I wasn't sure if these are "valid" given how far into the future they are, right? You're just creating times that are unlikely to expire during the lifetime of the execution of the app, right? Also, minor minor nit, but I think technically that `refreshExpires` should be larger than `identityExpires`.

Reply:
> You're just creating times that are unlikely to expire during the lifetime of the execution of the app, right?

Exactly.  This ensures that the UID2Identity will always be valid at the start of the test.  The values from the JSON file will eventually age out and could cause unintended issues for an app that only has one purpose.

> I think technically that `refreshExpires` should be larger than` identityExpires`.

Good call out.  I've now updated that to be the case.  👍